spec/exceptions_spec.rb

```diff
@@ -0,0 +1,21 @@
+require_relative "../lib/hathifiles_database/db/connection"
+
+RSpec.describe HathifilesDatabase::Exception do
+  describe HathifilesDatabase::Exception::WrongNumberOfColumns do
+    let(:test_params) { {htid: "test.001", count: 22, expected: 33} }
+
+    describe "#initialize" do
+      it "creates a WrongNumberOfColumns object" do
+        expect(described_class.new(**test_params)).to be_a HathifilesDatabase::Exception::WrongNumberOfColumns
```

Comment:
So this class will return an exception, but not raise it?
Just a tad confusing, because in my worldview, an `Exception` is something you raise, not return. I may absolutely be in the wrong here.

Reply:
See lib/hathifiles_database/linespec.rb:148. I think this is okay. I don't see a lot of `Exception.new` patterns but I think you have to call `new` if there are customized init parameters.